File: 3055-maximum-odd-binary-number/maximum-odd-binary-number.cpp
class Solution {
public:
    string maximumOddBinaryNumber(string s) {
        int n=s.size();
        int i=n-2;
        int k=0;
        while(i>=k){
            if(s[i]=='1' && s[n-1]!='1'){
                swap(s[i],s[n-1]);
            }
            else if(s[i]=='1'){
                while(s[k]=='1' && k<i){
                    k++;
                }
                swap(s[i],s[k]);
                k++;
            }
            i--;
        }
        return s;
    }
};

File: 3055-maximum-odd-binary-number/maximum-odd-binary-number.java
public class Solution {
    public String maximumOddBinaryNumber(String s) {
        char[] chars = s.toCharArray();
        int n = chars.length;
        int i = n - 2;
        int k = 0;
        while (i >= k) {
            if (chars[i] == '1' && chars[n - 1] != '1') {
                char temp = chars[i];
                chars[i] = chars[n - 1];
                chars[n - 1] = temp;
            } else if (chars[i] == '1') {
                while (chars[k] == '1' && k < i) {
                    k++;
                }
                char temp = chars[i];
                chars[i] = chars[k];
                chars[k] = temp;
                k++;
            }
            i--;
        }
        return new String(chars);
    }
}
